server/service/src/stocktake/update.rs

```diff
@@ -71,6 +73,43 @@ fn check_snapshot_matches_current_count(
     None
 }
 
+fn check_stock_lines_reduced_to_zero(
+    connection: &StorageConnection,
+    stocktake_lines: &Vec<StocktakeLine>,
+) -> Result<Option<Vec<StockLine>>, RepositoryError> {
+    let mut lines_reduced_to_zero = Vec::new();
+
+    for line in stocktake_lines {
+        let stock_line_row = match &line.stock_line {
+            Some(stock_line) => stock_line,
+            None => continue,
+        };
+        if let Some(counted_number_of_packs) = line.line.counted_number_of_packs {
+            let adjustment = stock_line_row.total_number_of_packs - counted_number_of_packs;
```

Comment:
This is the only place with this logic though? 🤔

Reply:
https://github.com/openmsupply/open-msupply/blob/763e9a609f8b349a820466ced48b0fbb05123395/server/service/src/stocktake/update.rs#L87-L92

here 

https://github.com/openmsupply/open-msupply/blob/763e9a609f8b349a820466ced48b0fbb05123395/server/service/src/stocktake_line/validate.rs#L103-L113

and here

https://github.com/openmsupply/open-msupply/blob/763e9a609f8b349a820466ced48b0fbb05123395/server/service/src/stocktake/update.rs#L214

in fact I think my nervousness might be validated, even thought I think the logic should have been fine as per this comment: https://github.com/openmsupply/open-msupply/pull/1178#discussion_r1106384256, looks like stock line total is used vs stocktake line snapshot (sorry I mean snapshot not total for stocktake line).

we use `delta` to do actual adjustment, which is stocktake_line.snapshot and stocktake_line.counted difference, but in the validation with use stock_line.total and stocktake_line.difference. Which would mean if total was lower then snapshot, validation may pass but the adjustment may still lead to negative (looks like this is guarded by `SnapshotCountCurrentCountMismatch`, although it does mean you can create stocktake line, then create and finalise outbound shipment with that line, and then update stocktake line without error and finalising stocktake straight away would lead to error) <- not requesting any change, but I would do an example of how I think the risk of this can be reduced in follow up PR